src/model/entities/device.ts

```diff
@@ -1,9 +1,95 @@
 import { StatusCode } from '../enums/status-code'
 
-export interface Device {
-  id: string
-  name: string
-  isConnected: boolean
-  statusCode: StatusCode
-  statusMessage: string
+export abstract class Device {
+  public type: string = 'Device'
+  public _id: string
+  private _name: string
+  private _isConnected: boolean
+  private _statusCode: StatusCode
+  private _statusMessage: string
+
+  constructor(id: string, name: string, isConnected: boolean, statusCode: StatusCode, statusMessage: string) {
```

Comment:
This is a lot of constructor parameters.
I think it might be safe to say, that we only need `id` and `name` for the constructor?

Sidenote: It might be that this should be an interface instead of a class, but I can't say for sure yet.